packages/chakra-components/src/components/Election/Election.tsx

```diff
@@ -114,18 +176,22 @@ export const useElectionProvider = ({
     // map questions back to expected Vote[] values
     const mapped = election.questions.map((q, k) => parseInt(values[k.toString()], 10))
 
+    let vote = new Vote(mapped)
+    setVoteInstance(vote)
+    if (typeof beforeSubmit === 'function' && !beforeSubmit(vote)) {
+      return
+    }
+
     try {
-      const vote = new Vote(mapped)
-      if (typeof beforeSubmit === 'function' && !beforeSubmit(vote)) {
-        return
+      let vid
+      if (censusType == CensusType.CSP) {
+        await cspAuthAndVote()
+      } else if (censusType == CensusType.WEIGHTED) {
+        vid = await weightedVote(vote)
+        setVoted(vid)
+        setVotesLeft(votesLeft - 1)
+        setIsAbleToVote(isInCensus && votesLeft - 1 > 0)
```

Comment:
Why declare `vid` as `let` there at the top instead of `const` when it's not used outside of this latest scope?

Reply:
Yeah, refactoring and forgot.